examples/link.rs

```diff
@@ -0,0 +1,70 @@
+use hassle_rs::*;
+use hassle_rs::utils::DefaultIncludeHandler;
+use rspirv::binary::Disassemble;
+
+fn main() {
+    let dxc = Dxc::new(None).unwrap();
+    let compiler = dxc.create_compiler().unwrap();
+    let library = dxc.create_library().unwrap();
+    let spirv = false;
+
+    let args = &if spirv {
+        vec!["-spirv", "-fspv-target-env=vulkan1.1spirv1.4"]
+    } else {
+        vec![]
+    };
+
+    let exports = compiler.compile(
+        &library.create_blob_with_encoding_from_str(include_str!("exports.hlsl")).unwrap(),
+        "exports.hlsl",
+        "",
+        "lib_6_6",
+        args,
+        Some(&mut DefaultIncludeHandler {}),
+        &[],
+    );
+    let use_exports = compiler.compile(
+        &library.create_blob_with_encoding_from_str(include_str!("use-export.hlsl")).unwrap(),
+        "use-exports.hlsl",
+        "",
+        "lib_6_6",
+        args,
+        Some(&mut DefaultIncludeHandler {}),
+        &[],
+    );
+
+    let exports = exports.ok().unwrap().get_result().unwrap();
+    let use_exports = use_exports.ok().unwrap().get_result().unwrap();
```

Comment:
Ugh it really is time for me to PR that error handling simplification, this `Result<DxcOperationResult, (DxcOperationResult, HRESULT)>` is so useless :grimacing: 

I think you even need to call `get_status()` etc :(

Reply:
Agreed, but I didn't feel like cleaning it up in this PR. There's a bit of an ambiguity there too because there are multiple possible failure operations making this quite complex. (e.g. link and compile can fail without an actual compiler or linker error)

Reply:
Yeah definitely not. I was more so thinking of the other way around: any of these might return an error code (including `.compile()` or `.link()`), but `DxcOperationResult` _might_ still contain an error blob so that there's more context _why_ the `HRESULT` was false. It just doesn't make sense that there's _also_ `get_status()` giving you two more `HRESULT`s _on top of the function return_.

Reply:
I think we can probably replace all of this with a `thiserror` type enum that just represents all of these error cases, and potentially just unwraps the DxcOperationResult into this new enum.